memory/src/main/scala/filodb.memory/format/vectors/Histogram.scala

```diff
@@ -324,6 +327,91 @@ final case class MaxHistogram(innerHist: MutableHistogram, max: Double) extends
   }
 }
 
+final case class DDSHistogram(relativeAccuracy: Double) extends HistogramWithBuckets {
+  def serialize(intoBuf: Option[MutableDirectBuffer] = None): MutableDirectBuffer = ???
+
+  val logGamma: Double = Math.log((1 + relativeAccuracy) / (1 - relativeAccuracy))
+  val maxIndexedValue = maxIndexableValue
+  val minIndexedValue = minIndexableValue
+  var zeroCount = 0
+  var indexMapping = new CubicallyInterpolatedMapping(relativeAccuracy)
+  var store = new UnboundedSizeDenseStore
+  final def bucketValue(no: Int): Double = 0
+  def buckets: HistogramBuckets = ???
+
+  def getCount: Long = zeroCount + store.getTotalCount
+
+  override def quantile(q: Double): Double = {
+    getValueAtQuantile(q, getCount)
+  }
+
+  def getValueAtQuantile(quantile: Double, count: Long): Double = {
+    if (quantile < 0 || quantile > 1) throw new IllegalArgumentException("The quantile must be between 0 and 1.")
+    if (count == 0) throw new NoSuchElementException
+    val rank: Long = (quantile * (count - 1)).toLong
+    if (rank < zeroCount) return 0
+    var bin: Bin = null
+    if (quantile <= 0.5) {
+      val binIterator: util.Iterator[Bin] = store.getAscendingIterator
+      var n: Long = zeroCount
+      do {
+        bin = binIterator.next
+        n += bin.getCount
+      } while ( {
+        n <= rank && binIterator.hasNext
+      })
+    }
+    else {
+      val binIterator: util.Iterator[Bin] = store.getDescendingIterator
+      var n: Long = count
+      do {
+        bin = binIterator.next
+        n -= bin.getCount
+      } while ( {
+        n > rank && binIterator.hasNext
+      })
+    }
+    indexMapping.value(bin.getIndex)
+  }
+
+  def compare(other: DDSHistogram): Int = {
```

Comment:
This seems to only compare logGamma and not the buckets?  Not sure of the purpose, and why we return an Int (as opposed to Boolean)